### PR TITLE
support raft learner

### PR DESCRIPTION
**Pull request by Connor1996**

### CODE REVIEW

File: proto/pdpb.proto [hunk 1]
```diff
@@ -240,6 +240,10 @@ message RegionHeartbeatRequest {
     uint64 keys_read = 9;
     // Approximate region size.
     uint64 approximate_size = 10;
+    // Pengding Learner peers are the non-voter peers that are catching up with leader's log
+    repeated metapb.Peer pending_learner_peers = 11;
```
Comment (hicqu) [inline proto/pdpb.proto]: How about `pending_learners`?

File: proto/pdpb.proto [hunk 2]
```diff
@@ -240,6 +240,10 @@ message RegionHeartbeatRequest {
...
+    // Pengding Learner peers are the non-voter peers that are waiting to be promoted to voter
+    repeated metapb.Peer complete_learner_peers = 12;
```
Comment (hicqu) [inline proto/pdpb.proto]: Does we really need this? We can got these learners by `setdiff(region.learners, pending_learners)`.

Comment (hicqu) [reply proto/pdpb.proto]: And I think you can also add `repeated Peer learners` in `metapb::Region` in this PR. @BusyJay what do you think?

**Comment by hicqu** [comment]
rest LGTM. @BusyJay @siddontang PTAL thanks.

File: proto/pdpb.proto (hunk 2 above)
Comment (Connor1996) [reply proto/pdpb.proto]: get it.

**Comment by hicqu** [comment]
LGTM.

**Comment by hicqu** [comment]
ping @BusyJay

**Comment by hicqu** [comment]
@Connor1996 Please re-geneerate code for resolving conflicts.

**Comment by hicqu** [comment]
@overvenus @BusyJay PTAL thanks.

**Comment by BusyJay** [comment]
LGTM, but I suggest not to merge it now. Let's develop base on this branch, and check what need to be tweaked.

**Comment by hicqu** [comment]
According to latest talk, we can remove `ConfChangeType::PromoteLearnerNode`. Please improve that.

**Comment by BusyJay** [comment]
I think we can merge this first, as no broken change is introduced.

**Pull request merged**